File: apps/backend/src/main/java/com/digisphere/pontofacil/dtos/company/CompanyDTO.java
package com.digisphere.pontofacil.dtos.company;

import com.digisphere.pontofacil.domain.adress.Address;
import com.digisphere.pontofacil.domain.company.Company;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record CompanyDTO(
  Long id,

  @NotBlank
  @Min(3)
  String name,

  String image,

  @NotNull
  Address address
) {

  public Company toEntity() {
    return new Company(
      id,
      name,
      image,
      address
    );
  }

}

File: apps/backend/src/main/kotlin/com/digisphere/pontofacil/company/api/CompanyDTO.kt
package com.digisphere.pontofacil.company.api

import com.digisphere.pontofacil.company.domain.Address
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

data class CompanyDTO(
  val id: Long? = null,

  @field:NotBlank
  @field:Size(min = 3)
  val name: String,

  val image: String,

  @field:NotNull
  val address: Address
)
